pkg/metrics/metrics.go

```diff
@@ -26,30 +26,43 @@ import (
 // Statistic is a common interface for histogram metrics
 type Statistic interface {
 	Collect() prometheus.Collector
-	EvaluateDuration(method string) func()
+	EvaluateDuration(labels prometheus.Labels) func()
+	EvaluateDurationForMethod(method string) func()
```

Comment:
These method names look a little confusing to me. Can we rename?
```
EvaluateDurationForMethod -> EvaluateMethodDuration
EvaluateDurationForType -> EvaluateMethodDurationWithType
```

Reply:
These function are not intent to measure method execution duration. They can be used for these purpose also, but this is only one of possible use scenarios. These functions are simple helpers, they do not Evaluate Method Duration. 

EvaluateDurationForMethod -> helps to evaluate duration of something and labeling it with label "method". EvaluateDurationAndAddLableMethod is the most precise name, but it is obviously too long and ugly.